ccmlib/scylla_node.py

```diff
@@ -379,16 +379,16 @@ def stop_scylla_manager_agent(self, gently):
                 pass
 
     def _wait_java_up(self, ip_addr, jmx_port):
-        start_time = time.time()
-        while (time.time() - start_time) < 30:
+        def wait_java_up_logic():
             with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as _socket:
                 _socket.settimeout(1.0)
                 try:
                     _socket.connect((ip_addr, jmx_port))
                     return True
                 except (socket.timeout, ConnectionRefusedError):
-                    time.sleep(0.05)
-        return False
+                    return False
+
+        return wait_for(func=wait_java_up_logic, timeout=30)
```

Comment:
Since if I remember correctly wait_for's default step is a whole second, please specify step=0.05 or you'll return the annoying oversleep issue we had here previously.